### PR TITLE
DashboardSceneSerializerLike: Update interface to support initial version and snapshot url

### PR DESCRIPTION
**Pull request by dprokop**
This extends the `DashboardSceneSerializerLike` interface to allow schema-independent dashboard version and snashot url retrieval, so that we do not neeed to check the schema version when those are used.

extracted from https://github.com/grafana/grafana/pull/97953

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx [hunk 1]
```diff
@@ -642,9 +642,18 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
...
+    return this._serializer.getInitialVersion();
+  };
+
+  public getSnapshotUrl = () => {
+    return this._serializer.getSnapshotUrl();
+  };
+
   /** Hacky temp function until we refactor transformSaveModelToScene a bit */
-  public setInitialSaveModel(saveModel?: Dashboard | DashboardV2Spec) {
+  public setInitialSaveModel(version: number | undefined, saveModel?: Dashboard | DashboardV2Spec) {
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/scene/DashboardScene.tsx:654]: Why don't we just save the metadata from the schema? We're losing the uid or the folder for the dashboard, for example.

Actually, the scene already save the meta. Can't we use the version from there?

**Review by ivanortegaalba — COMMENTED** [review]
I used `isNew` from meta because the version was not available: https://github.com/grafana/grafana/pull/98153

Maybe, we could just use the `scene.meta` from the scene, so we avoid to pass just the version. wdyt¿

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/scene/DashboardScene.tsx:654]: Ah you are right, not sure why I assumed that meta in state won't be updated after save (and it actually isn't, think it's a bug. the initial save model is updated, but not the meta).. Think we should be able to use it.

Comment (dprokop) [reply public/app/features/dashboard-scene/scene/DashboardScene.tsx]: actually, version is part of the meta, but it was not exposed in the `DashboardMeta` type, i've added it and updated the implementation. However, let's talk today about if it's even needed, i have doubts about `ignoreChanges` implementation.

**Review by ivanortegaalba — APPROVED** [review]
Thanks for applying the feedback 🙏

**Comment by ivanortegaalba** [comment]
@dprokop I pushed to the wrong branch 😓 I reverted it

**Pull request merged**